packages/shared_preferences/shared_preferences_tool/lib/src/shared_preferences_state.dart

```diff
@@ -0,0 +1,255 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:flutter/foundation.dart';
+
+import 'async_state.dart';
+
+@immutable
+
+/// A class that represents the state of the shared preferences tool.
+class SharedPreferencesState {
+  /// Default constructor for [SharedPreferencesState].
+  const SharedPreferencesState({
+    required this.allKeys,
+    required this.selectedKey,
+    this.editing = false,
+  });
+
+  /// A list of all keys in the shared preferences of the target debug session.
+  final List<String> allKeys;
+
+  /// The user selected key and its value in the shared preferences
+  /// of the target debug session.
+  final SelectedSharedPreferencesKey? selectedKey;
+
+  /// Whether the user is editing the value of the selected key.
+  final bool editing;
+
+  /// Creates a copy of this [SharedPreferencesState] but replacing the given
+  /// fields with the new values.
+  SharedPreferencesState copyWith({
+    List<String>? allKeys,
+    SelectedSharedPreferencesKey? selectedKey,
+    bool? editing,
+  }) {
+    return SharedPreferencesState(
+      allKeys: allKeys ?? this.allKeys,
+      selectedKey: selectedKey ?? this.selectedKey,
+      editing: editing ?? this.editing,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    return identical(this, other) ||
+        (other is SharedPreferencesState &&
+            listEquals(other.allKeys, allKeys) &&
```

Comment:
Well, it depends on what we want.

[1,2,3] != [3,2,1] for an actual list, it would be the same for a set. 

In the end it is irrelevant here, this is only being used for the tests matching anyway.